src/v/cluster/archival/ntp_archiver_service.cc

```diff
@@ -893,9 +893,13 @@ ss::future<std::error_code> ntp_archiver::process_anomalies(
   model::timestamp scrub_timestamp,
   std::optional<model::offset> last_scrubbed_offset,
   cloud_storage::scrub_status status,
-  cloud_storage::anomalies detected) {
+  cloud_storage::anomalies detected,
+  ss::abort_source& caller_as) {
+    ssx::composite_abort_source cas{caller_as, _as};
+    cas.as().check();
+
     // If there's ongoing housekeeping job, let it finish first.
-    auto units = co_await _mutex.get_units(_as);
+    auto units = co_await _mutex.get_units(cas.as());
```

Comment:
you might consider removing the `check()`. the semantics here are technically checked as you'd get an abort_requested rather than semaphore_aborted exception. the later is a subclass of the former, so probably no big deal, but maybe easier to just not need to think about it?